apps/condo/domains/acquiring/schema/RegisterMultiPaymentForOneReceiptService.js

```diff
@@ -0,0 +1,320 @@
+    },
+    BILLING_INTEGRATION_ORGANIZATION_CONTEXT_IS_DELETED: {
+        mutation: 'registerMultiPaymentForOneReceipt',
+        variable: ['data', 'receipt', 'id'],
+        code: BAD_USER_INPUT,
+        type: BILLING_INTEGRATION_ORGANIZATION_CONTEXT_IS_DELETED,
+        message: 'BillingIntegrationOrganizationContext is deleted for provided BillingReceipt',
+    },
+    ACQUIRING_INTEGRATION_DOES_NOT_SUPPORTS_BILLING_INTEGRATION: {
+        mutation: 'registerMultiPaymentForOneReceipt',
+        variable: ['data', 'receipt', 'id'],
+        code: BAD_USER_INPUT,
+        type: ACQUIRING_INTEGRATION_DOES_NOT_SUPPORTS_BILLING_INTEGRATION,
+        message: 'AcquiringIntegration does not supports following BillingReceipt\'s BillingIntegration: {unsupportedBillingIntegration}',
+    },
+    RECEIPT_HAS_DELETED_BILLING_INTEGRATION: {
+        mutation: 'registerMultiPaymentForOneReceipt',
+        variable: ['data', 'receipt', 'id'],
+        code: BAD_USER_INPUT,
+        type: RECEIPT_HAS_DELETED_BILLING_INTEGRATION,
+        message: 'BillingReceipt has deleted BillingIntegration',
+    },
+}
+
+const SENDER_FIELD_CONSTRAINTS = {
+    ...JSON_STRUCTURE_FIELDS_CONSTRAINTS,
+    dv: {
+        numericality: {
+            noStrings: true,
+            equalTo: 1,
+        },
+    },
+}
+
+const RegisterMultiPaymentForOneReceiptService = new GQLCustomSchema('RegisterMultiPaymentForOneReceiptService', {
+    types: [
+        {
+            access: true,
+            type: 'input RegisterMultiPaymentForOneReceiptInput { dv: Int!, sender: SenderFieldInput!, receipt: BillingReceiptWhereUniqueInput!, acquiringIntegrationContext: AcquiringIntegrationContextWhereUniqueInput! }',
+        },
+        {
+            access: true,
+            type: 'type RegisterMultiPaymentForOneReceiptOutput { dv: Int!, multiPaymentId: String!, webViewUrl: String!, feeCalculationUrl: String!, directPaymentUrl: String!, getCardTokensUrl: String! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canRegisterMultiPaymentForOneReceipt,
+            schema: 'registerMultiPaymentForOneReceipt(data: RegisterMultiPaymentForOneReceiptInput!): RegisterMultiPaymentForOneReceiptOutput',
+            resolver: async (parent, args, context) => {
+                // wrap validator function to the current call context
+                const { data } = args
+                const {
+                    dv,
+                    sender,
+                    receipt,
+                    acquiringIntegrationContext,
+                } = data
+
+                // Stage 0. Check if input is valid
+                if (dv !== 1) {
+                    throw new GQLError(errors.DV_VERSION_MISMATCH, context)
+                }
+
+                const senderErrors = validate(sender, SENDER_FIELD_CONSTRAINTS)
+                if (senderErrors && Object.keys(senderErrors).length) {
```

Comment:
`const senderErrorKeys = Object.keys(senderErrors)` and then reuse.

Reply:
It will throw an error, in case if senderErrors eq undefined or null. I prefer to keep it as is (exactly like in RegisterMultiPaymentService